src/state/hooks.ts

```diff
@@ -22,10 +22,11 @@ import { fetchWalletNfts } from './collectibles'
 export const useFetchPublicData = () => {
   const dispatch = useAppDispatch()
   const { slowRefresh } = useRefresh()
+  const { currentBlock } = useBlock()
   useEffect(() => {
     dispatch(fetchFarmsPublicDataAsync())
-    dispatch(fetchPoolsPublicDataAsync())
-  }, [dispatch, slowRefresh])
+    dispatch(fetchPoolsPublicDataAsync(currentBlock))
+  }, [dispatch, slowRefresh, currentBlock])
```

Comment:
Adding the current block here invalidates the necessity for the `slowRefresh`. Is this necssary?

Reply:
I keep the slowrefresh and remove the useblock otherwise we have to send requests unnecessarily every 6 seconds instead of 60. I am getting latest block number in useEffect directly